lammps-tutorials.tex

```diff
@@ -1256,15 +1214,27 @@ \subsubsection{Improving the script}
 \subsection{Tutorial 2: Pulling on a carbon nanotube}
 \label{carbon-nanotube-label}
 
-In this tutorial, the system of interest is a small carbon nanotube
-(CNT) in an empty box (Fig.\,\ref{fig:CNT}). An external force is imposed on
-the CNT, and its deformation is measured over time. To illustrate the
-difference between classical and reactive force fields, this tutorial is
-divided into two parts. Within the first part, a classical force field is
-used and the bonds between the atoms of the CNT are unbreakable. Within the
-second part, a reactive force field (named AIREBO \cite{stuart2000reactive})
-is used, allowing for the breaking of chemical bonds when the CNT undergoes
-strong deformation.
+% "classical" or rather "conventional molecular"?
+In this tutorial, the system of interest is a small, single walled
+carbon nanotube (CNT) in an empty box (Fig.\,\ref{fig:CNT}).  An
+% FIXME: not a force but strain.
+external force is imposed on the CNT, and its deformation is monitored
+over time.  To illustrate the difference between classical and reactive
+force fields, this tutorial is divided into two parts.  In the first
+part, a classical force field \cite{typelabel_paper} is used and the
+bonds between the atoms of the CNT are unbreakable.  In the second part,
+a reactive force field (named AIREBO \cite{stuart2000reactive}) is used,
+allowing for the breaking of chemical bonds when the CNT experiences
+large strain from the deformation.
+
+To set up this tutorial, select ``Start Tutorial 2'' from the
+``Tutorials'' menu of LAMMPS--GUI and follow the instructions.  This
+will select, and -- if needed -- create, a folder, and place several
+files into it.  The initial input file, set up for a single point energy
+calculation, will also be loaded into the editor under the name
+\textit{unbreakable.lmp}.  Additional files are a data file with a CNT
+\textit{cnt\_molecular.data} topology and geometry, and a file
+\textit{parameters.inc} with force field parameters.
```

Comment:
Following our recent email exchange, if we go with your naming style (which I am OK with), shouldn't it be unbreakable.parm instead of parameters.inc ?

Reply:
actually, we could also use unbreakable.data ?

Reply:
I intend to use the same data file for both parts, unbreakable and breakable and just use delete_bonds to strip away the unwanted parts. Apart from having to use `atom_style molecular` instead of `atomic` there is no difference and that also helps for the example of using fix bond/break and fix bond/create for visualization purposed. Now that I think about it, the "unbreakable" example could probably be made "properly breakable" with fix bond/react.

Same goes for the parameter include, the use of which is now justified as being more generic than for just one example. If anything it should thus be something like cnt_oplsaa.inc. But then again, without type labels this kind of include file is not really usable beyond a simple example, so we could just copy the contents into the input.
Following tutorial 1, we could also add "write_data unbreakable.data" right after the include and then change "unbreakable.lmp" to read the new data file with the Coeffs sections added and the include no longer needed.

There are many options and I am not firm on my choices. So think it over and let me know which you like best. Or put differently which is more important: consistency, simplicity, or showcasing many use cases?

Reply:
> Same goes for the parameter include, the use of which is now justified as being more generic than for just one example. If anything it should thus be something like cnt_oplsaa.inc. But then again, without type labels this kind of include file is not really usable beyond a simple example, so we could just copy the contents into the input.

Yes, its true, for this tutorial at least it would be OK to have the parameters within the inputs.

Reply:
> I intend to use the same data file for both parts, unbreakable and breakable and just use delete_bonds to strip away the unwanted parts.

In the first version of this tutorial, 2 different data files were used on purpose to illustrate that the number of columns in the Atom section must be changed with the atom styles, and that for AIREBO there is no need for bond sections. If this is something we don't want to illustrate after all, then yes using the same input has some merits as well as it simplifies the process...

Reply:
> There are many options and I am not firm on my choices. So think it over and let me know which you like best. Or put differently which is more important: consistency, simplicity, or showcasing many use cases?

I prefer consistency for my personal use, but for tutorials, simplicity is probably the way to go

Reply:
We can have the two different data files, but then it would be IMO helpful that the name reflect the atom style: cnt_molecular.data and cnt_atomic.data (instead of cnt_atom.data). I would then add a sentence to look at the comment after the "Atoms" section title which should reflect the atom style needed. This comment is not required but written by compliant tools (LAMMPS, TopoTools, haven't checked OVITO) and if present LAMMPS will print a warning that the atom style does not match.

Sadly, the data file format is not self-descriptive and thus dependent on external settings, which makes dealing with them in tools like VMD and OVITO quite difficult at times. I've been complaining about this a lot, but we never found the time or energy to do something about it. In part because the writing of the file is deeply entangles into the various styles and thus a *lot* of would would be required to support some different file format.